rust/signer/src/lib.rs

```diff
@@ -229,6 +229,48 @@ export! {
         let signature: Vec<u8> = signature.from_hex().ok()?;
         keypair.verify_signature(&message, &signature)
     }
+
+    @Java_io_parity_signer_EthkeyBridge_ethkeyDecryptDataRef
+    fn decrypt_data_ref(data: &str, password: String) -> Option<i64> {
+        let password = Protected::new(password.into_bytes());
+        let crypto: Crypto = serde_json::from_str(data).ok()?;
+        let decrypted = crypto.decrypt(&password).ok()?;
+        let res = Box::into_raw(Box::new(String::from_utf8(decrypted).ok())) as i64; 
+        Some(res)
+    }
+
+    @Java_io_parity_signer_EthkeyBridge_ethkeyDestroyDataRef
+    fn destroy_data_ref(data_ref: i64) -> () {
+        unsafe { Box::from_raw(data_ref as *mut String) };
+    }
+
+    @Java_io_parity_signer_EthkeyBridge_ethkeyBrainwalletSignWithRef
+    fn ethkey_brainwallet_sign_with_ref(seed_ref: i64, message: &str) -> Option<String> {
+        let seed = unsafe { Box::from_raw(seed_ref as *mut String) };
```

Comment:
If the seed did not got by the pointer, will be a panic here? 

Is there any way to catch the error here?

Reply:
> If the seed did not got by the pointer, will be a panic here?
Yeah this is unsafe. If we destroy the pointer or pass the wrong number this behavior will be undefined.
> Is there any way to catch the error here?
No it will most likely just segfault. We have to prevent it on the typescript side with a safe abstraction.